### PR TITLE
ES6 refactor desc file utils

### PR DESCRIPTION
**Pull request by pllearns**
- All tests currently passing. 
- linting passes.

- Awaiting feedback on further refactoring needed.

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	"extends": ["eslint:recommended", "plugin:node/recommended"],
 	"env": {
 		"node": true,
-		"es6": true
+    "es6": true
```
Comment (sokra) [inline .eslintrc]: spacing

File: lib/DescriptionFileUtils.js [hunk 2]
```diff
@@ -1,31 +1,35 @@
+"use strict";
```
Comment (sokra) [inline lib/DescriptionFileUtils.js]: use strict should be after the license comment

File: lib/DescriptionFileUtils.js [hunk 3]
```diff
@@ -1,31 +1,35 @@
...
+const assign = require("object-assign"); //eslint-disable-line
+const forEachBail = require("./forEachBail");
+
+const loadDescriptionFile = (resolver, directory, filenames, callback) => {
 	(function findDescriptionFile() {
 		forEachBail(filenames, function(filename, callback) {
 			var descriptionFilePath = resolver.join(directory, filename);
 			if(resolver.fileSystem.readJson) {
-				resolver.fileSystem.readJson(descriptionFilePath, function(err, content) {
+				resolver.fileSystem.readJson(descriptionFilePath, ((err, content) => {
```
Comment (sokra) [inline lib/DescriptionFileUtils.js]: the brackets around the arrow function are not neccessary

File: lib/DescriptionFileUtils.js [hunk 4]
```diff
@@ -1,31 +1,35 @@
...
 					if(err) {
 						if(typeof err.code !== "undefined") return callback();
 						return onJson(err);
 					}
 					onJson(null, content);
-				});
+				}));
 			} else {
-				resolver.fileSystem.readFile(descriptionFilePath, function(err, content) {
+				resolver.fileSystem.readFile(descriptionFilePath, ((err, content) => {
```
Comment (sokra) [inline lib/DescriptionFileUtils.js]: same

File: lib/DescriptionFileUtils.js [hunk 5]
```diff
@@ -42,7 +46,7 @@ function loadDescriptionFile(resolver, directory, filenames, callback) {
 					path: descriptionFilePath
 				});
 			}
-		}, function(err, result) {
+		}, ((err, result) => {
```
Comment (sokra) [inline lib/DescriptionFileUtils.js]: same

File: .eslintrc (hunk 1 above)
Comment (pllearns) [inline .eslintrc]: The GitHub spacing here is off. I double checked on my editor and it is spaced the same as `node`.

File: lib/DescriptionFileUtils.js (hunk 3 above)
Comment (pllearns) [inline lib/DescriptionFileUtils.js]: I may have to refactor more if I remove the brackets. I get an `unexpected token if` `parsing error`

File: .eslintrc (hunk 1 above)
Comment (mikesherov) [inline .eslintrc]: @pllearns you changed tabs to spaces here. Please revert the changes to this file.

File: lib/DescriptionFileUtils.js (hunk 3 above)
Comment (mikesherov) [inline lib/DescriptionFileUtils.js]: Maybe that's a problem with your editor? The extra parens are definitely unnecessary

File: lib/DescriptionFileUtils.js [hunk 6]
```diff
@@ -1,31 +1,35 @@
+"use strict";
+
 /*
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var forEachBail = require("./forEachBail");
 
-function loadDescriptionFile(resolver, directory, filenames, callback) {
+const assign = require("object-assign"); //eslint-disable-line
```
Comment (mikesherov) [inline lib/DescriptionFileUtils.js]: please remove this line.

**Comment by pllearns** [comment]
Thank you for the feedback @mikesherov @sokra many apologies for the delay on getting changes fixed. I should be able to get this done tomorrow.

**Comment by mikesherov** [comment]
@pllearns thanks for the updates! it's still a bit broken. Would you mind if I went in and made the remaining changes to this PR to get it to pass?

**Comment by pllearns** [comment]
thanks @mikesherov go ahead and make changes, could you ping me when you do? I want to know what I messed up on. Thanks!!

**Comment by ooflorent** [comment]
Thank you for your PR but it appears that the file was previously converted to ES6. I'm closing the PR for this reason.
